piecrust/src/store.rs

```diff
@@ -200,17 +224,65 @@ fn page_path<P: AsRef<Path>>(memory_dir: P, page_index: usize) -> PathBuf {
     memory_dir.as_ref().join(format!("{page_index}"))
 }
 
+fn page_path_main<P: AsRef<Path>, S: AsRef<str>>(
+    memory_dir: P,
+    page_index: usize,
+    commit_id: S,
+) -> io::Result<PathBuf> {
+    let commit_id = commit_id.as_ref();
+    let dir = memory_dir.as_ref().join(commit_id);
+    fs::create_dir_all(&dir)?;
+    Ok(dir.join(format!("{page_index}")))
+}
+
+fn index_path_main<P: AsRef<Path>, S: AsRef<str>>(
+    main_dir: P,
+    commit_id: S,
+) -> io::Result<PathBuf> {
+    let commit_id = commit_id.as_ref();
+    let dir = main_dir.as_ref().join(commit_id);
+    fs::create_dir_all(&dir)?;
+    Ok(dir.join(INDEX_FILE))
+}
+
+fn commit_id_to_hash<S: AsRef<str>>(commit_id: S) -> Hash {
+    let hash: [u8; 32] = hex::decode(commit_id.as_ref())
+        .expect("Hex decoding of commit id string should succeed")
+        .try_into()
+        .expect("Commit id string conversion should succeed");
```

Comment:
I understand that the `expect`s failing here means that the state is corrupt somehow. Any thoughts on how to deal with such a situation in general?

Reply:
I agree that we should have a sweep of all `expects` from the point of view of what kind of state corruption do we want to tolerate and how. This is a non-trivial subject so I'd rather leave it as a todo for the near future.